xds/internal/xdsclient/authority.go

```diff
@@ -306,6 +333,29 @@ func (a *authority) newConnectionError(err error) {
 	// from the transport layer.
 	for _, rType := range a.resources {
 		for _, state := range rType {
+			// If the connection/stream breaks after the resource has been requested,
+			// but before the response is received, we stop the timer here. When the stream
+			// is recreated, the transport layer will resend all previously requested resources,
+			// and we will start the timer in the `onResourceSend` callback.
+			if state.wState == watchStateRequested {
+				state.wTimer.Stop()
+				state.wState = watchStateStarted
+			}
+		}
+	}
+
+	// Do not propagate errors to watchers if error is of type IgnoredAdsErrorType,
+	// since this error occurred after a successful message was received on the stream.
+	if xdsresource.ErrType(err) == xdsresource.ErrorTypeIgnored {
+		a.logger.Debugf("ErrorTypeIgnored error was ignored and not sent to watchers.")
```

Comment:
More human readable log message, and this can be a warning.
`log.Warningf("Watchers not notified since ADS stream failed after having received at least one response: %v", err)`